### ISSUE TITLE
user-agent header missing in auth request

### ISSUE THREAD

**Issue by metallkopf** [containers/image#843]
the docker client authentication makes a http request through `getBearerToken` which doesnt sets the user-agent like `makeRequestToResolvedURLOnce`

![agentless](https://user-images.githubusercontent.com/47254555/76174710-e56cdd00-61a0-11ea-937d-e11103979555.png)

```diff
diff --git a/docker/docker_client.go b/docker/docker_client.go
index 7bfa3030c..b7183c129 100644
--- a/docker/docker_client.go
+++ b/docker/docker_client.go
@@ -588,6 +588,9 @@ func (c *dockerClient) getBearerToken(ctx context.Context, challenge challenge,
 	if c.username != "" && c.password != "" {
 		authReq.SetBasicAuth(c.username, c.password)
 	}
+	if c.sys != nil && c.sys.DockerRegistryUserAgent != "" {
+		authReq.Header.Add("User-Agent", c.sys.DockerRegistryUserAgent)
+	}
 	logrus.Debugf("%s %s", authReq.Method, authReq.URL.String())
 	res, err := c.client.Do(authReq)
 	if err != nil {
```

**Comment by sjhx** [comment]
As a registry service provider this would be helpful to identify the client creating incoming requests good or bad.

**Comment by mtrmac** [comment]
This was fixed by #897. Thanks!
